WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java

```diff
@@ -404,16 +404,26 @@ private void initViewModel() {
                         handleNewPageAction(PagePostCreationSourcesDetail.PAGE_FROM_MY_SITE);
                         break;
                 }
-
-                if (mMainBottomSheetFragment != null) {
-                    mMainBottomSheetFragment.dismiss();
-                    mMainBottomSheetFragment = null;
-                }
             });
 
             mFloatingActionButton.setOnClickListener(v -> {
-                mMainBottomSheetFragment = new MainBottomSheetFragment();
-                mMainBottomSheetFragment.show(getSupportFragmentManager(), MAIN_BOTTOM_SHEET_TAG);
+                MainBottomSheetFragment bottomSheet = new MainBottomSheetFragment();
+                bottomSheet.show(getSupportFragmentManager(), MAIN_BOTTOM_SHEET_TAG);
+                mViewModel.setIsBottomSheetShowing(true);
+            });
+
+            mViewModel.isBottomSheetShowing().observe(this, isShowing -> {
+                if (!isShowing) {
+                    FragmentManager fm = getSupportFragmentManager();
+                    if (fm != null) {
+                        MainBottomSheetFragment bottomSheet =
+                                (MainBottomSheetFragment) fm.findFragmentByTag(MAIN_BOTTOM_SHEET_TAG);
+
+                        if (bottomSheet != null) {
```

Comment:
What do you think about moving the construction of the fragment here? The logic should be something like:
```
if (isShowing && bottomSheet == null) {
          MainBottomSheetFragment bottomSheet = new MainBottomSheetFragment();
          bottomSheet.show(getSupportFragmentManager(), MAIN_BOTTOM_SHEET_TAG);
} else if (!isShowing && bottomSheet != null) {
          bottomSheet.dismiss();
}
```
this would make the `mFloatingActionButton.setOnClickListener` very simple (just `mViewModel.setIsBottomSheetShowing(true);`

Reply:
I was thinking about this initially since I also agree it cleans the click listener a lot, if I'm not wrong though in this way we need an instance variable (like mBottomSheet) to keep track of the null condition and I preferred to go the other way. Anyway I'm 50/50 between them so if you prefer the other way round or if I'm misunderstanding what you were suggesting let me know and I will just change it 👍.

Reply:
Don't know why but I totally misunderstood your suggestion 🤦‍♂️! Forget my previous comment! Changed here 80284ee